src/netbase.cpp

```diff
@@ -572,6 +702,20 @@ bool GetProxy(enum Network net, proxyType &proxyInfoOut) {
     return true;
 }
 
+/**
+ * Set the proxy to use for populating our peer list using a one-shot `getaddr`
```

Comment:
I believe most of this comment is incorrect, or at least very misleading :(.

SetNameProxy makes sure that all resolves are routed through a proxy rather than being done ourselves. No need to bring the higher-level 'getaddr' into it.

Maybe you grepped for HasNameProxy and missed the use of GetNameProxy in ConnectNode()? If so, looking at that may change your understanding somewhat.

Reply:
Okay this might also inform my documentation on `net.cpp` proper. I see that in the "Resolve" stage of `ConnectNode`, we are guaranteed a non-external query lookup if we have a name proxy. However, if `pszDest` was a hostname, the `Lookup` would fail, and we drop down to the `else if (pszDest && GetNameProxy(proxy))` case in the "Connect" stage, where we connect to the host through the proxy without doing any resolution ourselves (and delegate the resolution to the proxy). Is that a correct understanding?

Reply:
That matches what I remember about this logic. The reason is that if we're using Tor, we don't want to perform DNS lookups ourselves (as those would leak to DNS servers, instead of over the proxy). Instead, we do a name-based connection where we give the name directly as host to connect to the proxy (which in the case of Tor will cause the exit node to do the resolving locally and establish a connection with the result).

Reply:
Reworded here: https://github.com/bitcoin/bitcoin/pull/15824/files#diff-4942713bcd2752bb531bf76c9a0ba38eR714